src/predicate/str/regex.rs

```diff
@@ -0,0 +1,40 @@
+use regex;
+
+use Predicate;
+
+/// An error that occurred during parsing or compiling a regular expression.
+pub type RegexError = regex::Error;
+
+/// Predicate that uses regex matching
+///
+/// This is created by the `predicate::str::is_match`.
+#[derive(Clone, Debug)]
+pub struct RegexPredicate {
+    re: regex::Regex,
+}
+
+impl Predicate for RegexPredicate {
+    type Item = str;
+
+    fn eval(&self, variable: &str) -> bool {
+        self.re.is_match(variable)
+    }
+}
+
+/// Creates a new `Predicate` that diffs two strings.
```

Comment:
Should actually be something like "applies a regular expression match to a string"